tests/test_wallet_wallet_recovery.py

```diff
@@ -71,29 +66,144 @@ def _assert_recovered(
             assert old_wallet.safe_chains == wallet.safe_chains
             assert wallet.decrypt_mnemonic(password) == mnemonics[wallet.ledger_type]
 
+    @staticmethod
+    def _assert_recovery_requirements(
+        wallet_manager: MasterWalletManager,
+        prepare_json: t.Dict,
+        backup_owner: str,
+        recovery_requirements: t.Dict[str, t.Any],
+        is_refill_required: bool = False,
+    ) -> None:
+        balances = recovery_requirements["balances"]
+        total_requirements = recovery_requirements["total_requirements"]
+        refill_requirements = recovery_requirements["refill_requirements"]
+
+        expected_balances = ChainAmounts()
+        expected_requirements = ChainAmounts()
+        expected_pending_bo_swaps: t.Dict = {}
+
+        for wallet in wallet_manager:
+            new_wallet_json = next(
+                (
+                    w
+                    for w in prepare_json["wallets"]
+                    if w["new_wallet"]["ledger_type"] == wallet.ledger_type.value
+                )
+            )["new_wallet"]
+            for chain, safe in wallet.safes.items():
+                chain_str = chain.value
+                expected_balances.setdefault(chain_str, {}).setdefault(
+                    backup_owner, {}
+                ).setdefault(ZERO_ADDRESS, 0)
+                expected_balances[chain_str][backup_owner][
+                    ZERO_ADDRESS
+                ] = recovery_requirements["balances"][chain_str][backup_owner][
+                    ZERO_ADDRESS
+                ]
+                expected_requirements.setdefault(chain_str, {}).setdefault(
+                    backup_owner, {}
+                ).setdefault(ZERO_ADDRESS, 0)
+
+                ledger_api = get_default_ledger_api(chain)
+                owners = get_owners(ledger_api=ledger_api, safe=safe)
+                if new_wallet_json["address"] not in owners:
+                    expected_requirements[chain_str][backup_owner][
+                        ZERO_ADDRESS
+                    ] += DEFAULT_RECOVERY_TOPUPS[chain][ZERO_ADDRESS]
+                    expected_pending_bo_swaps.setdefault(chain_str, [])
+                    if safe not in expected_pending_bo_swaps[chain_str]:
+                        expected_pending_bo_swaps[chain_str].append(safe)
+
+                balance = balances[chain_str][backup_owner][ZERO_ADDRESS]
+                requirement = total_requirements[chain_str][backup_owner][ZERO_ADDRESS]
+                shortfall = refill_requirements[chain_str][backup_owner][ZERO_ADDRESS]
+                assert shortfall == max(requirement - balance, 0)
+
+                if is_refill_required:
+                    assert shortfall > 0
+                else:
+                    assert shortfall == 0
```

Comment:
do we check anywhere with `is_refill_required = True`?

Reply:
yeah I mean we're not testing any case where `expected_is_refill_required` is True, right?

Reply:
So far no.